membership_product_set/tests/test_membership.py

```
@@ -0,0 +1,127 @@
+            'property_account_receivable_id': False,
+        })
+        cls.gold_product = cls.env['product.product'].create({
+            'type': 'service',
+            'name': 'Membership Gold',
+            'membership': True,
+            'membership_date_from': fields.Date.today(),
+            'membership_date_to': cls.next_month,
+            'list_price': 100.00,
+            'taxes_id': [(4, cls.tax_15p.id, False)]
+        })
+        cls.silver_product = cls.env['product.product'].create({
+            'type': 'service',
+            'name': 'Membership Silver',
+            'membership': True,
+            'membership_date_from': fields.Date.today(),
+            'membership_date_to': cls.next_month,
+            'list_price': 50.00,
+        })
+        cls.basic_product = cls.env['product.product'].create({
+            'type': 'service',
+            'name': 'Membership Basic',
+            'membership': True,
+            'membership_date_from': fields.Date.today(),
+            'membership_date_to': cls.next_month,
+            'list_price': 20.00,
+        })
+        cls.product_set = cls.env['product.product'].create({
+            'type': 'service',
+            'name': 'Membership Set',
+            'membership': True,
+            'membership_set': True,
+            'membership_date_from': fields.Date.today(),
+            'membership_date_to': cls.next_month,
+            'list_price': 150.00,
+            'membership_set_products': [
+                (4, cls.gold_product.id, False),
+                (4, cls.silver_product.id, False),
+            ]
+        })
+
+    def test_membership_invoice(self):
+        wizard = self.env['membership.invoice'].with_context(
+            active_ids=[self.partner.id],
+            active_model='res.partner'
+        ).create({
+            'product_id': self.product_set.id,
+            'member_price': 150,
+        })
+        with self.assertRaises(UserError):
```

Consider using  **assertRaisesRegexp**, it can avoid to catch another **UserError** not intended and swallow the failure. 

` with self.assertRaisesRegexp(UserError, r'^My expected message')`

---

Thanks for the tip. I will improve it now.